app/domain/people_skills_filter.rb

```diff
@@ -1,21 +1,45 @@
 # frozen_string_literal: true
 
 class PeopleSkillsFilter
-  attr_reader :entries, :rated, :level, :interest
+  attr_reader :entries, :rated, :levels_and_interests_for_skills
 
-  def initialize(entries, rated, level = nil, interest = nil)
+  # rubocop:disable Metrics/ParameterLists
+  def initialize(entries, rated, levels = [], interests = [], skill_ids = [])
     @entries = entries
     @rated = rated
-    @level = level
-    @interest = interest
+    @levels_and_interests_for_skills = []
+    set_levels_and_interests_for_skills(levels, interests, skill_ids)
   end
+  # rubocop:enable Metrics/ParameterLists
 
   def scope
     filter_by_level_and_interest(filter_by_rated)
   end
 
   private
 
+  def set_levels_and_interests_for_skills(levels, interests, skill_ids)
+    return if levels.blank? || skill_ids.blank? || interests.blank?
```

Comment:
How about `return unless levels || skill_ids || interests` , should work the same

Reply:
unless levels || skill_ids || interests is equivalent to if !levels && !skill_ids && !interests. This condition will be true if levels, skill_ids, and interests are all nil or false. It will be false if any of these variables are non-nil or non-false.

if levels.blank? || skill_ids.blank? || interests.blank? checks if levels, skill_ids, and interests are all empty or nil. It will be true if any of these variables are nil, an empty string, or an empty collection (such as an empty array or hash).